### DISCUSSION TITLE
Setting up GDB Debugger with ChampSim

### DISCUSSION THREAD

**Discussion by bagelbagelbeagle** [ChampSim/ChampSim#557] [Q&A] [answered]
I was wondering what is the recommended way of building ChampSim for debugging. I noticed that the `make test` rule includes the debug flags needed for the g++ compiler, but it seems to go through a non-configurable test routine. I was thinking of creating my own separate `debug` rule in the makefile, but I was wondering if there's another recommended method I'm missing. I'm going to use the gdb debugger for this. Thanks for the help!

**Comment by ngober** [comment, marked as answer]
Usually when I'm debugging, I'll just pass the flags directly to make: `make CXXFLAGS='-g -Og'`. You could create a rule, but it would probably just be this.
